### PR TITLE
tests: use a known test phone for SNS v3 tests

### PR DESCRIPTION
**Pull request by david-luna**
Reuse phone number defined on `aws-sdk` tests in `@aws-sdk/clilent-sns`.  

### Checklist

- [ ] Implement code
- [x] Modify tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [ ] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js
Comment (trentm) [inline test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js:1]: Hunting around a bit, I found that the range 555-0100 through 555-0199 is reserved for fictional usage.

https://en.wikipedia.org/wiki/555_(telephone_number)
> Only 555-0100 through 555-0199 are now specifically reserved for fictional use; the other numbers have been reserved for actual assignment.

So, let's use `+1-555-555-0100` for our fictional number.

There are some hits in other files too:

```
test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.mjs
26:    PhoneNumber: '+34555555555',

test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js
113:    PhoneNumber: '1-555-555-5555',
180:    PhoneNumber: `+34${TEST_IGNORE_PHONE}`,
193:    PhoneNumber: '1-555-555-5555',

test/instrumentation/modules/aws-sdk/sns.test.js
106:          PhoneNumber: '1-555-555-5555',
109:      '1-555-555-5555',
173:          PhoneNumber: '15037299028',
192:          PhoneNumber: '15555555555',
106:          PhoneNumber: '1-555-555-5555',
```

**Review by trentm — CHANGES_REQUESTED** [review]

File: test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js
Comment (trentm) [reply test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js:1]: > So, let's use `+1-555-555-0100` for our fictional number.

That value is used in an AWS docs example here: https://docs.aws.amazon.com/cli/latest/reference/sns/publish.html#examples

Comment (david-luna) [reply test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js:1]: I'd like that but I'm getting this message with the fictional number

```
InvalidParameterException: Invalid parameter: PhoneNumber Reason: +1-555-555-0100 is not valid to publish to
    at de_InvalidParameterExceptionRes (/Users/david/Documents/repos/el/apm-agent-nodejs/node_modules/@aws-sdk/client-sns/dist-cjs/protocols/Aws_query.js:2449:23)
    at de_PublishCommandError (/Users/david/Documents/repos/el/apm-agent-nodejs/node_modules/@aws-sdk/client-sns/dist-cjs/protocols/Aws_query.js:1704:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/david/Documents/repos/el/apm-agent-nodejs/node_modules/@smithy/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
```

I guess there is a validation on the client to not use fictional numbers.

Comment (david-luna) [reply test/instrumentation/modules/@aws-sdk/fixtures/use-client-sns.js:1]: Turns out the client does not parse the number so it does not accept a formatted value with the character `-`.
Changing the number to `+15555550100` works :)

**Review by trentm — APPROVED** [review]

**Pull request merged**